pyteal/ast/itxn.py

```diff
@@ -425,7 +482,7 @@ def MethodCall(
             # Set the fields for the app call in app args and foreign arrays
             *fields_to_set,
             # Add any remaining fields specified by the user
-            InnerTxnBuilder.SetFields(extra_fields),
+            InnerTxnBuilder.SetFields(cast(dict, extra_fields)),
```

Comment:
I don't think this is the right way to do:
```suggestion
            InnerTxnBuilder.SetFields({} if extra_fields is None else extra_fields),
```

The mutable default argument in python is somewhat notorious, see https://docs.python-guide.org/writing/gotchas/ as a small example. In our case, we do not internally change stuffs to such argument, but it is a code pattern to avoid in python.